File: COM/Samples/Dot Net/VS2005/Visual Basic/Sample2/Form1.vb
Imports System.Drawing
Imports System.Drawing.Printing

Public Class Form1
    Inherits System.Windows.Forms.Form

#Region " Vom Windows Form Designer generierter Code "

    Public Sub New()
        MyBase.New()
        InitializeComponent()
    End Sub

    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    Private components As System.ComponentModel.IContainer

    Friend WithEvents Button1 As System.Windows.Forms.Button
    Friend WithEvents Button2 As System.Windows.Forms.Button
    Friend WithEvents Timer1 As System.Windows.Forms.Timer
    Friend WithEvents TextBox1 As System.Windows.Forms.TextBox
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container
        Me.Button1 = New System.Windows.Forms.Button
        Me.Timer1 = New System.Windows.Forms.Timer(Me.components)
        Me.TextBox1 = New System.Windows.Forms.TextBox
        Me.Button2 = New System.Windows.Forms.Button
        Me.SuspendLayout()
        '
        'Button1
        '
        Me.Button1.Enabled = False
        Me.Button1.Location = New System.Drawing.Point(24, 8)
        Me.Button1.Name = "Button1"
        Me.Button1.Size = New System.Drawing.Size(152, 40)
        Me.Button1.TabIndex = 0
        Me.Button1.Text = "&Start"
        '
        'Button2
        '
        Me.Button2.Enabled = False
        Me.Button2.Location = New System.Drawing.Point(288, 8)
        Me.Button2.Name = "Button2"
        Me.Button2.Size = New System.Drawing.Size(152, 40)
        Me.Button2.TabIndex = 2
        Me.Button2.Text = "&Preview"
        '
        'TextBox1
        '
        Me.TextBox1.BackColor = System.Drawing.Color.FromArgb(CType(255, Byte), CType(255, Byte), CType(192, Byte))
        Me.TextBox1.Location = New System.Drawing.Point(0, 56)
        Me.TextBox1.Multiline = True
        Me.TextBox1.Name = "TextBox1"
        Me.TextBox1.ReadOnly = True
        Me.TextBox1.ScrollBars = System.Windows.Forms.ScrollBars.Both
        Me.TextBox1.Size = New System.Drawing.Size(464, 80)
        Me.TextBox1.TabIndex = 1
        Me.TextBox1.Text = "TextBox1"
        Me.TextBox1.WordWrap = False
        '
        'Form1
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(466, 142)
        Me.Controls.Add(Me.Button1)
        Me.Controls.Add(Me.Button2)
        Me.Controls.Add(Me.TextBox1)
        Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle
        Me.MaximizeBox = False
        Me.MinimizeBox = False
        Me.Name = "Form1"
        Me.Text = "Sample2 - PDFCreator COM interface"
        Me.ResumeLayout(False)

    End Sub

#End Region

    Private WithEvents _PDFCreator As PDFCreator.clsPDFCreator
    Private pErr As PDFCreator.clsPDFCreatorError

    Private pd As PrintDocument

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim parameters As String
        AddStatus("Status: Program is started.", True)

        pErr = New PDFCreator.clsPDFCreatorError
        _PDFCreator = New PDFCreator.clsPDFCreator

        parameters = "/NoProcessingAtStartup"

        If _PDFCreator.cStart(parameters) = True Then
            _PDFCreator.cClearCache()
            _PDFCreator.cOption("UseAutosave") = 0
            Button1.Enabled = True
            Button2.Enabled = True
            _PDFCreator.cPrinterStop = False
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        pd = New PrintDocument
        AddHandler pd.PrintPage, AddressOf pd_PrintPage
        pd.PrinterSettings.PrinterName = "PDFCreator"
        pd.DocumentName = "PDFCreator Dot Net - Sample2"
        pd.Print()
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        pd = New PrintDocument
        AddHandler pd.PrintPage, AddressOf pd_PrintPage
        pd.PrinterSettings.PrinterName = "PDFCreator"
        pd.DocumentName = "PDFCreator Dot Net - Sample2"
        Dim ppdlg As New System.Windows.Forms.PrintPreviewDialog
        With ppdlg
            .Document = pd
            .WindowState = FormWindowState.Maximized
            .ShowDialog()
            .Dispose()
        End With
    End Sub

    Private Sub pd_PrintPage(ByVal sender As Object, ByVal ev As PrintPageEventArgs)
        Dim x As Single, y As Single, r As Single
        With pd.PrinterSettings.DefaultPageSettings.PaperSize
            x = .Width / 2
            y = .Height / 2
            r = .Width / 4
        End With
        DrawCircles(x - r / 2, y - r / 2, r, 5, ev)
    End Sub

    Private Sub DrawCircles(ByVal x As Single, ByVal y As Single, ByVal r As Single, ByVal rec As Long, ByVal ev As PrintPageEventArgs)
        If rec <> 0 Then
            Dim p As New Pen(Color.Red)
            With ev.Graphics
                .DrawString("PDFCreator", New Font("Arial", 16), Brushes.Black, 100, 100, New StringFormat)
                .DrawEllipse(p, x - r, y, r, r)
                .DrawEllipse(p, x + r, y, r, r)
                .DrawEllipse(p, x, y - r, r, r)
                .DrawEllipse(p, x, y + r, r, r)
                .DrawEllipse(p, x, y, r, r)
            End With
            DrawCircles(x - r / 2, y - r / 2, r / 2, rec - 1, ev)
            DrawCircles(x - r / 2, y + r, r / 2, rec - 1, ev)
            DrawCircles(x + r, y - r / 2, r / 2, rec - 1, ev)
            DrawCircles(x + r, y + r, r / 2, rec - 1, ev)
        End If
    End Sub

    Private Sub _PDFCreator_eReady() Handles _PDFCreator.eReady
        AddStatus("Status: """ & _PDFCreator.cOutputFilename & """ was created!")
        _PDFCreator.cPrinterStop = True
    End Sub

    Private Sub _PDFCreator_eError() Handles _PDFCreator.eError
        pErr = _PDFCreator.cError
        AddStatus("Status: Error[" & pErr.Number & "]: " & pErr.Description)
    End Sub

    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        Timer1.Enabled = False
    End Sub

    Private Sub AddStatus(ByVal Str1 As String, Optional ByVal ClearStatus As Boolean = False)
        With TextBox1
            If ClearStatus = True Then
                .Text = Str1
                .SelectionStart = 0
            Else
                If .Text.Length = 0 Then
                    .Text = Str1
                    .SelectionStart = 0
                Else
                    .Text = .Text & vbCrLf & Str1
                End If
            End If
        End With
    End Sub

    Private Sub Form1_Closing(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs) Handles MyBase.Closing
        _PDFCreator.cClose()
        System.Runtime.InteropServices.Marshal.ReleaseComObject(_PDFCreator)
        System.Runtime.InteropServices.Marshal.ReleaseComObject(pErr)
        pErr = Nothing
        GC.Collect()
        GC.WaitForPendingFinalizers()
        GC.Collect()
        GC.WaitForPendingFinalizers()
    End Sub
End Class

File: COM/Samples/Dot Net/VS2005/C#/Sample2/Form1.cs
using System;
using System.Collections;
using System.Drawing;
using System.Drawing.Printing;
using System.Windows.Forms;

namespace Sample2
{
	public class Form1 : System.Windows.Forms.Form
	{
		private PDFCreator.clsPDFCreator _PDFCreator;
		private PDFCreator.clsPDFCreatorError pErr;

		private PrintDocument pd;
		
		private System.Windows.Forms.Button button1;
		private System.Windows.Forms.Button button2;
		private System.Windows.Forms.TextBox textBox1;
		private System.Windows.Forms.Timer timer1;
		private System.ComponentModel.IContainer components;

		public Form1()
		{
			InitializeComponent();
		}

		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Vom Windows Form-Designer generierter Code
		private void InitializeComponent()
		{
			this.components = new System.ComponentModel.Container();
			this.button1 = new System.Windows.Forms.Button();
			this.button2 = new System.Windows.Forms.Button();
			this.textBox1 = new System.Windows.Forms.TextBox();
			this.timer1 = new System.Windows.Forms.Timer(this.components);
			this.SuspendLayout();
			// 
			// button1
			// 
			this.button1.Enabled = false;
			this.button1.Location = new System.Drawing.Point(16, 8);
			this.button1.Name = "button1";
			this.button1.Size = new System.Drawing.Size(152, 40);
			this.button1.TabIndex = 0;
			this.button1.Text = "&Start";
			this.button1.Click += new System.EventHandler(this.button1_Click);
			// 
			// button2
			// 
			this.button2.Enabled = false;
			this.button2.Location = new System.Drawing.Point(288, 8);
			this.button2.Name = "button2";
			this.button2.Size = new System.Drawing.Size(152, 40);
			this.button2.TabIndex = 1;
			this.button2.Text = "&Preview";
			this.button2.Click += new System.EventHandler(this.button2_Click);
			// 
			// textBox1
			// 
			this.textBox1.BackColor = System.Drawing.Color.FromArgb(((System.Byte)(255)), ((System.Byte)(255)), ((System.Byte)(192)));
			this.textBox1.Location = new System.Drawing.Point(0, 56);
			this.textBox1.Multiline = true;
			this.textBox1.Name = "textBox1";
			this.textBox1.ReadOnly = true;
			this.textBox1.ScrollBars = System.Windows.Forms.ScrollBars.Both;
			this.textBox1.Size = new System.Drawing.Size(464, 80);
			this.textBox1.TabIndex = 2;
			this.textBox1.Text = "textBox1";
			this.textBox1.WordWrap = false;
			// 
			// Form1
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(466, 142);
			this.Controls.Add(this.button1);
			this.Controls.Add(this.button2);
			this.Controls.Add(this.textBox1);
			this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
			this.MaximizeBox = false;
			this.MinimizeBox = false;
			this.Name = "Form1";
			this.Text = "Sample2 - PDFCreator COM interface";
			this.Closing += new System.ComponentModel.CancelEventHandler(this.Form1_Closing);
			this.Load += new System.EventHandler(this.Form1_Load);
			this.ResumeLayout(false);

		}
		#endregion

		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		private void AddStatus(string Str1, bool ClearStatus)
		{
			if (ClearStatus)
			{
				textBox1.Text = Str1;
				textBox1.SelectionStart = 0;
			}
			else
			{
				if (textBox1.Text.Length == 0)
				{
					textBox1.Text = Str1;
					textBox1.SelectionStart = 0;
				}
				else
				{
					textBox1.Text = textBox1.Text + "\r\n" + Str1;
				}
			}
		}
 
		private void Form1_Load(object sender, System.EventArgs e)
		{
			string parameters;
			AddStatus("Status: Program is started.", true);

			pErr = new PDFCreator.clsPDFCreatorError();

			_PDFCreator = new PDFCreator.clsPDFCreator();
			_PDFCreator.eError  += new PDFCreator.__clsPDFCreator_eErrorEventHandler(_PDFCreator_eError); 
			_PDFCreator.eReady  += new PDFCreator.__clsPDFCreator_eReadyEventHandler(_PDFCreator_eReady); 
			
			parameters = "/NoProcessingAtStartup";

			if (_PDFCreator.cStart(parameters, false))
			{
				button1.Enabled = true;
				button2.Enabled = true;
				_PDFCreator.cClearCache();
				_PDFCreator.set_cOption("UseAutosave", 0);
				_PDFCreator.cPrinterStop = false;
			}
		}

		private void _PDFCreator_eReady()
		{
			AddStatus("Status: \"" + _PDFCreator.cOutputFilename + "\" was created!", false);
			_PDFCreator.cPrinterStop = true;
		}

		private void _PDFCreator_eError()
		{
			pErr = _PDFCreator.cError;
			AddStatus("Status: Error[" + pErr.Number + "]: " + pErr.Description, false);
		}

		private void button1_Click(object sender, System.EventArgs e)
		{
			pd = new PrintDocument();
			pd.PrintPage += new PrintPageEventHandler(pd_PrintPage);
			pd.PrinterSettings.PrinterName = "PDFCreator";
			pd.DocumentName = "PDFCreator Dot Net - Sample2";
			pd.Print();
		}

		private void button2_Click(object sender, System.EventArgs e)
		{
			pd = new PrintDocument();
			pd.PrintPage += new PrintPageEventHandler(this.pd_PrintPage);
			pd.PrinterSettings.PrinterName = "PDFCreator";
			pd.DocumentName = "PDFCreator Dot Net - Sample2";
			PrintPreviewDialog ppdlg = new PrintPreviewDialog();
			ppdlg.Document = pd;
			ppdlg.WindowState = FormWindowState.Maximized;
			ppdlg.ShowDialog();
			ppdlg.Dispose();
		}

		private void pd_PrintPage(object sender, PrintPageEventArgs ev)
		{
			float x, y, r;
			x = pd.PrinterSettings.DefaultPageSettings.PaperSize.Width / 2; 
			y = pd.PrinterSettings.DefaultPageSettings.PaperSize.Height / 2; 
			r = pd.PrinterSettings.DefaultPageSettings.PaperSize.Width / 4; 
			DrawCircles(x - r / 2, y - r / 2, r, 5, ev);
		}
		private void DrawCircles(float x, float y, float r, long rec, PrintPageEventArgs ev)
		{
			if (rec != 0)
			{
				Pen p = new Pen(Color.Red);
				ev.Graphics.DrawString("PDFCreator", new Font("Arial", 16), Brushes.Black, 100, 100, new StringFormat());
				ev.Graphics.DrawEllipse(p, x - r, y, r, r);
				ev.Graphics.DrawEllipse(p, x + r, y, r, r);
				ev.Graphics.DrawEllipse(p, x, y - r, r, r);
				ev.Graphics.DrawEllipse(p, x, y + r, r, r);
				ev.Graphics.DrawEllipse(p, x, y, r, r);
				DrawCircles(x - r / 2, y - r / 2, r / 2, rec - 1, ev);
				DrawCircles(x - r / 2, y + r, r / 2, rec - 1, ev);
				DrawCircles(x + r, y - r / 2, r / 2, rec - 1, ev);
				DrawCircles(x + r, y + r, r / 2, rec - 1, ev);
			}
		}

		private void Form1_Closing(object sender, System.ComponentModel.CancelEventArgs e)
		{
			_PDFCreator.cClose();
            while (_PDFCreator.cProgramIsRunning)
            {
                Application.DoEvents();
                System.Threading.Thread.Sleep(100);
            }
			pErr = null;
            _PDFCreator = null;
			GC.Collect();
            GC.WaitForPendingFinalizers();
        }
	}
}